content/ember-4-0-released.md

```diff
@@ -0,0 +1,233 @@
+- The internal APIs of `<LinkTo>`, `<Input>`, `<Checkbox>`, and `<Textarea>` are now private,
+  and subclassing them is no longer supported.
+  An example of subclassing looks like this: `export class MyLinkComponent extends LinkTo`.
+  Apps or addons that subclass can install
+  the library [`@ember/legacy-built-in-components`](https://github.com/emberjs/ember-legacy-built-in-components)
+  as a stepping stone, following [this deprecation guide](https://deprecations.emberjs.com/v3.x#toc_ember-built-in-components-import).
+- ...and additional uncommon deprecations found in the [Ember.js 3.x deprecation
+  guide](https://deprecations.emberjs.com/v3.x)
+
+Many of these removed APIs date back to Ember 1.x, and are rarely used now (or should be rarely used).
+
+For more details on the changes in Ember.js 4.0, please review the
+[Ember.js 4.0.0 release page](https://github.com/emberjs/ember.js/releases/tag/v4.0.0).
+
+### Deprecations added in 4.0
+
+The following public API, marked as deprecated in Ember 4.0, will be removed in
+Ember 5.0:
+
+- Use of `Ember.assign` is deprecated. You should replace any calls to `Ember.assign` with `Object.assign` or use the object spread operator. See the [deprecation guide](https://deprecations.emberjs.com/v4.x#toc_ember-polyfills-deprecate-assign) for examples.
+
+## Changes in Ember Data 4.0
+
+Ember Data is the official data persistence library for Ember.js applications.
+
+This release removes APIs that were [deprecated in the 3.x cycle](https://deprecations.emberjs.com/ember-data/v3.x). Examples include:
+
+- The API `store.defaultAdapter` is removed - [deprecation guide](https://deprecations.emberjs.com/ember-data/v3.x#toc_ember-data-default-adapter)
+- Support for relying on fallback behavior for adapter type is removed.
+  You should [specify adapter types explicitly](https://deprecations.emberjs.com/ember-data/v3.x#toc_ember-data-default-adapter).
+- Similarly, `adapter.defaultSerializer` and support for relying on fallback behavior for the
+  serializer type [is removed](https://deprecations.emberjs.com/ember-data/v3.x#toc_ember-data-default-serializers)
+- The [Evented API is removed](https://deprecations.emberjs.com/ember-data/v3.x#toc_evented-api-usage)
+- ...and other APIs listed in the [deprecation guide](https://deprecations.emberjs.com/ember-data/v3.x)
+
+For more details on the changes in Ember Data 4.0, please review the
+[Ember Data 4.0.0 release page](https://github.com/emberjs/data/releases/tag/v4.0.0).
+
+## Changes in Ember CLI 4.0
```

Comment:
Maybe (some of) these might be worth including in the Ember CLI section?

- https://github.com/ember-cli/ember-cli/pull/9563
- https://github.com/ember-cli/ember-cli/pull/9579
- https://github.com/ember-cli/ember-cli/pull/9595

Reply:
hmmm. Major versions are supposed to only be about removing deprecated APIs not adding new things. Were these really added in 4.0?

Reply:
Yes => https://github.com/ember-cli/ember-cli/blob/master/CHANGELOG.md#v400.

Reply:
I've updated the post to call out the CI provider change (also GitHub Actions is now the default in 4.0) and the exclude/include change.

I left out the pnpm PR as I'm not sure how much that is a step toward pnpm support vs "pnpm just works well everywhere now". I think the latter would be worth a call-out, but highlighting an incremental step might be a bit confusing.

I asked @Turbo87 for feedback on Discord. He could always update this in a followup.